watsononlinestore/watson_online_store.py

```diff
@@ -66,84 +76,95 @@ def parse_slack_output(self, slack_rtm_output):
-        if existing_user:
-            # return some error
-            return False
-
-        # start to create the customer, add to DB when complete
-        self.customer = OnlineStoreCustomer(email=email_addr)
-
-        return True
 
-    def handle_AddName(self):
-        """ Add User Name to existing OnlineStoreCustomer
+        email_addr = user_data['email']
+        first = user_data['first_name']
+        last = user_data['last_name']
+        self.customer = OnlineStoreCustomer(email=email_addr,
+                                            first_name=first,
+                                            last_name=last,
+                                            shopping_cart="",
+                                            logged_in=True)
+
+    def create_user_from_ui(self, user_json):
+        """Create a new user from data in Slack
+
+            Authenticated user in slack will have email, First, and Last
+           names. Create a user in the DB for this. Note that a different
+           UI will require different code here
         """
-        full_name = str(self.context['full_name'])
-        first, last = full_name.split()
-        self.customer.first_name = first
-        self.customer.last_name = last
-
-        # TODO continue to ask for favorites and remove next lines, maybe,
-        #  but not for now.
-        self.customer.logged_in = True
-        self.context['logged_in'] = False
-        # logged_in = {'logged_in': True}
-        # FIXME: logged_in undefined here. Need to finish/fix this.
-        # self.context = self.context_merge(self.context, logged_in)
-        if DEBUG:
-            print("AddName context:\n{}".format(self.context))
-        self.cloudant_online_store.add_customer_obj(self.customer)
 
-        return True
+        email_addr = user_json['user']['profile']['email']
+        first = user_json['user']['profile']['first_name']
+        last = user_json['user']['profile']['last_name']
+        self.customer = OnlineStoreCustomer(email=email_addr,
+                                            first_name=first,
+                                            last_name=last,
+                                            shopping_cart="",
+                                            logged_in=True)
+
+    def init_customer(self, user_id):
+        """ Get user from DB, or create entry for user.
+
+            Note that this is specific to using Slack as the UI.
+             A different UI will require different code for the API
+             calls.
+        """
+        user_json = None
+        user_data = None
+
+        try:
+            # Get the authenticated user profile from Slack
+            user_json = self.slack_client.api_call("users.info",
+                                                   user=user_id)
+        except:
+            pass
+        if DEBUG and user_json and "error" not in user_json:
+            print("user_from_slack:\n{}\n".format(user_json))
```

Comment:
Use "stuff \n%s\n" % params instead.  Because 1 - this looks ugly to me (but maybe I'm just wrong there and 2 - more importantly one of these blew up on an apostrophe (unicode care) but the typical "%s" % parm style handled it fine.